src/ray/core_worker/actor_handle.h

```diff
@@ -70,9 +70,13 @@ class ActorHandle {
 
   int64_t MaxTaskRetries() const { return inner_.max_task_retries(); }
 
+  bool IsPersistedToGCS() const { return inner_.is_persisted_to_gcs(); }
```

Comment:
I would make this flag a private member of `ActorHandle` instead of defining it as a member of `rpc::ActorHandle`. In general, we try to limit `rpc::ActorHandle` to the info that you need to call tasks on that actor, whereas the `is_persisted_to_gcs` flag can be set independently by each caller.

Reply:
Ah also, I think you will need to hold the lock for the read/write.

Reply:
I actually added this to inner because I didn't want to call GCS whenever actor handle is created. But, I just realized if we set the timer, this problem will be resolved 99.9%! I will move it to the private member.